yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,176 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    ExtractorError,
+    int_or_none,
+    str_or_none,
+    traverse_obj,
+    urlencode_postdata,
+)
+
+
+class FreeTvBaseIE(InfoExtractor):
+    def _get_api_response(self, content_id, postdata, resource_type):
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            content_id, data=urlencode_postdata(postdata), fatal=True,
+            note=f'Downloading {content_id} {resource_type} JSON')
+
+        if response.get('data') is False:
+            raise ExtractorError(f"Response doesn't contain {resource_type} data")
+
+        return response
+
+    def _extract_video(self, content_id, action='olyott_video_play'):
+        api_response = self._get_api_response(
+            content_id, resource_type='video',
+            postdata={
+                'action': action,
+                'contentID': content_id,
+            })
+
+        video_id = traverse_obj(api_response, ('data', 'displayMeta', 'contentID'), expected_type=str)
+        video_url = traverse_obj(api_response, ('data', 'displayMeta', 'streamURLVideo'), expected_type=str)
+
+        if determine_ext(video_url) != 'm3u8':
+            raise ExtractorError('Manifest was not found', video_id=video_id)
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, video_id, 'mp4')
```

Comment:
what's the point of this? Did you mean this?
```suggestion
        video_url = traverse_obj(api_response, ('data', 'displayMeta', 'streamURLVideo'), expected_type=str)

        formats, subtitles = [], {}
        if determine_ext(video_url) != 'm3u8':
            self.raise_no_formats('Manifest was not found', video_id=video_id)
            formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, video_id, 'mp4')
```